### PR TITLE
Change stanza create behavior

### PR DESCRIPTION
**Pull request by benjaminjb**
**Checklist:**
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change, as applicable?
 - [x] Have you tested your changes on all related environments with successful results, as applicable?
   - [ ] Have you added automated tests?

**Type of Changes:**
 - [ ] New feature
 - [ ] Bug fix
 - [ ] Documentation
 - [ ] Testing enhancement
 - [ ] Other

**What is the current behavior (link to any open issues here)?**

Stanza-create behavior is tied to certain conditions on the PG cluster. 

**What is the new behavior (if this is a feature change)?**
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

The operator will attempt a stanza-create command, and if that returns an error indicating a PG version mismatch, PGO will attempt a stanza-upgrade.

**Other Information**:
Issue [sc-13696]

### CODE REVIEW

File: internal/pgbackrest/pgbackrest.go [hunk 1]
```diff
@@ -36,11 +41,15 @@ type Executor func(
...
-// that prevented the pgBackRest stanza-create or stanza-upgrade command from running (with a
-// config mismatch indicating that the pgBackRest configuration as stored in the cluster's
-// pgBackRest ConfigMap has not yet propagated to the Pod).
+// depending on the boolean value of the "upgrade" function parameter. This function is invoked
+// by the "reconcileStanzaCreate" function with "upgrade" set to false; if the stanza already
+// exists but the PG version has changed, pgBackRest will error with the "errMsgBackupDbMismatch"
+// error. If that occurs, we then rerun the command with "upgrade" set to true.
+// - https://github.com/pgbackrest/pgbackrest/blob/release/2.38/src/command/check/common.c#L154-L156
+// If the bool returned from this function is false, this indicates that a pgBackRest config hash
+// mismatch was identified that prevented the pgBackRest stanza-create or stanza-upgrade command
```
Comment (cbandy) [inline internal/pgbackrest/pgbackrest.go:50]:
```suggestion
// If the bool returned from this function is true, this indicates that a pgBackRest config hash
// mismatch was identified that prevented the pgBackRest stanza-create or stanza-upgrade command
```

File: Makefile [hunk 2]
```diff
@@ -95,7 +95,7 @@ deletenamespaces:
 
 # Install the postgrescluster CRD
 install:
-	$(PGO_KUBE_CLIENT) apply -k ./config/crd
+	$(PGO_KUBE_CLIENT) apply --server-side=true --force-conflicts -k ./config/crd
```
Comment (tjmoore4) [inline Makefile:98]: :pick: 
Not a blocker, but perhaps this should go in a separate commit?

File: internal/pgbackrest/pgbackrest.go [hunk 3]
```diff
@@ -36,11 +41,15 @@ type Executor func(
...
-// depending on the boolean value of the "upgrade" function parameter.  If the bool returned from
-// this function is false, this indicates that a pgBackRest config hash mismatch was identified
-// that prevented the pgBackRest stanza-create or stanza-upgrade command from running (with a
-// config mismatch indicating that the pgBackRest configuration as stored in the cluster's
-// pgBackRest ConfigMap has not yet propagated to the Pod).
+// depending on the boolean value of the "upgrade" function parameter. This function is invoked
+// by the "reconcileStanzaCreate" function with "upgrade" set to false; if the stanza already
+// exists but the PG version has changed, pgBackRest will error with the "errMsgBackupDbMismatch"
+// error. If that occurs, we then rerun the command with "upgrade" set to true.
+// - https://github.com/pgbackrest/pgbackrest/blob/release/2.38/src/command/check/common.c#L154-L156
```
Comment (tjmoore4) [inline internal/pgbackrest/pgbackrest.go:48]: :+1: Nice link.

**Review by tjmoore4 — APPROVED** [review]
:+1: Looks good. One non-blocking nitpick that could be addressed.

File: Makefile (hunk 2 above)
Comment (benjaminjb) [reply Makefile]: Removed, I think we have another PR where this might fit more naturally.
